### PR TITLE
Fixed issues with custom LWIP compilation and C++11 standard.

### PR DESCRIPTION
**Pull request by slaff**
First reported in #946.

### CODE REVIEW

**Comment by slaff** [comment]
@ADiea @avr39-ripe Can you guys check if that PR is working on your Windows and FreeBSD-current respectively?

**Comment by anakod** [comment]
How to check it?

Is it related: https://github.com/SmingHub/Sming/issues/945 or it's old issue?

File: Sming/third-party/.patches/esp-open-lwip.patch [hunk 1]
```diff
@@ -144,3 +125,46 @@ index eaa8dd6..6a08192 100644
...
+diff --git a/include/lwip/mem.h b/include/lwip/mem.h
+index af6e360..6d8cabd 100644
+--- a/include/lwip/mem.h
++++ b/include/lwip/mem.h
+@@ -52,19 +52,19 @@ typedef size_t mem_size_t;
+  */
+ #ifndef MEMLEAK_DEBUG
+ #ifndef mem_free
+-#define mem_free vPortFree
++#define mem_free(s) vPortFree(s, "", 0)
```
Comment (ADiea) [inline Sming/third-party/.patches/esp-open-lwip.patch]: Why use empty string,0 when memleak is not defined? why not use simpler function version?

**Comment by ADiea** [comment]
Hi I don't understand reason of this PR
Were pv* functions not linking or? It compiles fine for me on windows

**Comment by slaff** [comment]
> Were pv* functions not linking or? It compiles fine for me on windows

@ADiea No, the include path was wrong and thus the compilation was not taking the correct header files. With this change it should be fixed. 

> It compiles fine for me on windows
Great, if it compiles on FreeBSD-current too then we are ready to merge it.

**Comment by ADiea** [comment]
Sorry, I mean that the OLD version is compiling fine on Windows, I cannot test right now the PR. Also I don't understand why not also define the simple functions and use the memory leak functions even if mem leak is not defined. These will be redefined in ummalloc.

**Comment by avr39-ripe** [comment]
On freebsd-current I got this error with this pull request checked in:
``` code
Installing libsming
(!) SSL support is not enabled. To enable it type: 'make clean; make ENABLE_SSL=1'
Done
Making spiffy utility
CC spiffy.c
In file included from /usr/local/lib/gcc48/gcc/x86_64-portbld-freebsd12.0/4.8.5/include-fixed/unistd.h:46:0,
                 from spiffy.c:2:
/usr/local/lib/gcc48/gcc/x86_64-portbld-freebsd12.0/4.8.5/include-fixed/sys/types.h:266:1: error: unknown type name '__vm_ooffset_t'
 typedef __vm_ooffset_t vm_ooffset_t;
 ^
/usr/local/lib/gcc48/gcc/x86_64-portbld-freebsd12.0/4.8.5/include-fixed/sys/types.h:268:1: error: unknown type name '__vm_pindex_t'
 typedef __vm_pindex_t vm_pindex_t;
 ^
gmake[4]: *** [Makefile:31: spiffy.o] Ошибка 1
gmake[3]: *** [Makefile:343: spiffy/spiffy] Ошибка 2
gmake[3]: выход из каталога «/usr/home/shurik/src/esp8266/Sming/Sming»
gmake[2]: *** [Makefile:410: third-party/esp-open-lwip/Makefile.open] Ошибка 2
gmake[2]: выход из каталога «/usr/home/shurik/src/esp8266/Sming/Sming»
gmake[1]: *** [Makefile:410: third-party/spiffs/makefile] Ошибка 2
gmake[1]: выход из каталога «/usr/home/shurik/src/esp8266/Sming/Sming»
gmake: *** [Makefile:410: third-party/rboot/Makefile] Ошибка 2
```

File: Sming/third-party/.patches/esp-open-lwip.patch (hunk 1 above)
Comment (slaff) [inline Sming/third-party/.patches/esp-open-lwip.patch]: What is your suggestion?

**Comment by slaff** [comment]
@avr39-ripe 

```
In file included from /usr/local/lib/gcc48/gcc/x86_64-portbld-freebsd12.0/4.8.5/include-fixed/unistd.h:46:0,
                 from spiffy.c:2:
/usr/local/lib/gcc48/gcc/x86_64-portbld-freebsd12.0/4.8.5/include-fixed/sys/types.h:266:1: error: unknown type name '__vm_ooffset_t'
 typedef __vm_ooffset_t vm_ooffset_t;
 ^
/usr/local/lib/gcc48/gcc/x86_64-portbld-freebsd12.0/4.8.5/include-fixed/sys/types.h:268:1: error: unknown type name '__vm_pindex_t'
 typedef __vm_pindex_t vm_pindex_t;
 ^
gmake[4]: *** [Makefile:31: spiffy.o] Ошибка 1
```
That looks like a different error, that should not be related to this PR. Just in case - can you get the latest develop version and try to compile Sming there?

**Comment by avr39-ripe** [comment]
Fix issue with spiffy compilation in #953

**Comment by avr39-ripe** [comment]
while try to build my rBoot-based app with recent changes got this
``` code
E2 /usr/home/shurik/src/esp8266/DragonMaster/out/firmware/rboot.bin
gmake[1]: выход из каталога «/usr/home/shurik/src/esp8266/Sming/Sming/third-party/rboot»
C+ app/app.cpp
In file included from /usr/home/shurik/src/esp8266/esp-open-sdk/sdk/include/user_interface.h:11:0,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/system/include/esp_systemapi.h:10,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/../Wiring/../include/user_config.h:29,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/../Wiring/WiringFrameworkDependencies.h:11,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/../Wiring/WiringFrameworkIncludes.h:11,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/SmingCore.h:13,
                 from lib/application/application.h:3,
                 from include/app.h:10,
                 from app/app.cpp:7:
/usr/home/shurik/src/esp8266/Sming/Sming/third-party/esp-open-lwip/include/lwip/ip_addr.h:213:23: error: invalid suffix on literal; C++11 requires a space between literal and string macro [-Werror=literal-suffix]
   LWIP_DEBUGF(debug, ("%"U16_F".%"U16_F".%"U16_F".%"U16_F,             \
                       ^
/usr/home/shurik/src/esp8266/Sming/Sming/third-party/esp-open-lwip/include/lwip/ip_addr.h:213:31: error: invalid suffix on literal; C++11 requires a space between literal and string macro [-Werror=literal-suffix]
   LWIP_DEBUGF(debug, ("%"U16_F".%"U16_F".%"U16_F".%"U16_F,             \
                               ^
/usr/home/shurik/src/esp8266/Sming/Sming/third-party/esp-open-lwip/include/lwip/ip_addr.h:213:40: error: invalid suffix on literal; C++11 requires a space between literal and string macro [-Werror=literal-suffix]
   LWIP_DEBUGF(debug, ("%"U16_F".%"U16_F".%"U16_F".%"U16_F,             \
                                        ^
/usr/home/shurik/src/esp8266/Sming/Sming/third-party/esp-open-lwip/include/lwip/ip_addr.h:213:49: error: invalid suffix on literal; C++11 requires a space between literal and string macro [-Werror=literal-suffix]
   LWIP_DEBUGF(debug, ("%"U16_F".%"U16_F".%"U16_F".%"U16_F,             \
                                                 ^
In file included from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/../Wiring/../include/user_config.h:29:0,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/../Wiring/WiringFrameworkDependencies.h:11,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/../Wiring/WiringFrameworkIncludes.h:11,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/SmingCore.h:13,
                 from lib/application/application.h:3,
                 from include/app.h:10,
                 from app/app.cpp:7:
/usr/home/shurik/src/esp8266/Sming/Sming/system/include/esp_systemapi.h:63:62: error: conflicting declaration of C function 'void* ets_memcpy(void*, const void*, size_t)'
 extern void *ets_memcpy(void *dest, const void *src, size_t n);
                                                              ^
In file included from /usr/home/shurik/src/esp8266/Sming/Sming/third-party/esp-open-lwip/include/lwip/arch.h:43:0,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/third-party/esp-open-lwip/include/lwip/debug.h:35,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/third-party/esp-open-lwip/include/lwip/opt.h:46,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/third-party/esp-open-lwip/include/lwip/ip_addr.h:35,
                 from /usr/home/shurik/src/esp8266/esp-open-sdk/sdk/include/user_interface.h:11,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/system/include/esp_systemapi.h:10,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/../Wiring/../include/user_config.h:29,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/../Wiring/WiringFrameworkDependencies.h:11,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/../Wiring/WiringFrameworkIncludes.h:11,
                 from /usr/home/shurik/src/esp8266/Sming/Sming/SmingCore/SmingCore.h:13,
                 from lib/application/application.h:3,
                 from include/app.h:10,
                 from app/app.cpp:7:
/usr/home/shurik/src/esp8266/Sming/Sming/third-party/esp-open-lwip/include/arch/cc.h:47:13: note: previous declaration 'void ets_memcpy(void*, const void*, uint32)'
 extern void ets_memcpy(void*, const void*, uint32);
             ^
cc1plus: all warnings being treated as errors
gmake: *** [/usr/home/shurik/src/esp8266/Sming/Sming/Makefile-rboot.mk:537: out/build/app/app.o] Ошибка 1
```

**Pull request merged**